File: 169/main.py
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        """
        approach
        loop through list, count elements in a map, if any appears more than n // 2, return
        test cases:
        []
        [1]
        [1,1]
        [1,1,2]
        [1,2,2]
        """
        counts = {}
        threshold = len(nums) // 2
        for item in nums:
            if item not in counts:
                counts[item] = 0
            counts[item] += 1
            if counts[item] > threshold:
                return item

File: 169/main.ts
function majorityElement(nums: number[]): number {
  let counts = new Map();
  for (let num of nums) {
    if (!counts.has(num)) counts.set(num, 0);
    counts.set(num, counts.get(num) + 1);
    if (counts.get(num) > Math.floor(nums.length / 2)) return num;
  }
}

// have a map of counts
// for each char, increase counts. if counts of any char >= n/2, return
